packages/react-components/src/components/Dialpad/Dialpad.tsx

```diff
@@ -0,0 +1,237 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { concatStyleSets, IStyle, ITextFieldStyles, mergeStyles, TextField, useTheme } from '@fluentui/react';
+import React, { useState } from 'react';
+import { useLocale } from '../../localization';
+import { containerStyles, digitStyles, rowStyles, subStyles, textFieldStyles } from './Dialpad.styles';
+
+/**
+ * Strings of {@link Dialpad} that can be overridden.
+ *
+ * @public
+ */
+export interface DialpadStrings {
+  errorText: string;
+  defaultText: string;
+}
+
+/**
+ * Props for {@link Dialpad} component.
+ *
+ * @public
+ */
+export interface DialpadProps {
+  strings?: DialpadStrings;
+  containerStyles?: IStyle;
+  rowStyles?: IStyle;
+  textFieldStyles?: Partial<ITextFieldStyles>;
+  digitStyles?: IStyle;
+  subStyles?: IStyle;
+}
+
+/**
+ * A component to allow users to enter phone number through clicking on dialpad/using keyboard
+ *
+ *
+ * @public
+ */
+export const Dialpad = (props: DialpadProps): JSX.Element => {
+  const localeStrings = useLocale().strings.dialpad;
+  const strings = { ...localeStrings, ...props.strings };
+  return <DialpadContainer errorText={strings.errorText} defaultText={strings.defaultText} {...props} />;
+};
+
+const formatPhoneNumber = (value: string, defaultText: string): string => {
+  // if input value is default text then just return
+  if (value === defaultText) {
+    return value;
+  }
+
+  // if input value is falsy eg if the user deletes the input, then just return
+  if (!value) {
+    return value;
+  }
+
+  // clean the input for any non-digit values.
+  const phoneNumber = value.replace(/[^\d*#+]/g, '');
```

Comment:
Can this be fixed to support characters like `(` and `-`, e.g. a user should be able to type (555) 555-5555

Reply:
Is it a requirement for this to do input scrubbing? Concerned other countries/languages may have input we didn't consider

Reply:
This is in the design but I am planning on discussing it in the upcoming feature meeting. On my phone right now it formats numbers from any country into the North American format. The input scrubbing here allows numbers and * and # and +, which are the input options from the dialpad

Reply:
I am thinking if we should allow user to type in ( and ) since if they type (555) 555-5555, auto format will mess it up. Or should we turn off autofomat when we detect ( and - in the input? Standard dialpad on the phone does not allow input for () and - since they are not on the dialpad

Reply:
On android if I copy and paste a number they show up:
![image](https://user-images.githubusercontent.com/2684369/167487688-d1b54bf1-dcf0-4eb8-beb6-80136f3e092e.png)

Reply:
Oh apologies you're right its the auto formatter!

Reply:
Sounds good, I think given its in design discuss what the support should be and this can be updated in the future 👍

Reply:
The system probably cleared the original brackets and inserted new brackets when formatting. Just tried pasting pure brackets into my phone, they are not showing up